saas/api/metrics.py

```diff
@@ -837,3 +837,230 @@ def paginate_queryset(self, queryset):
         page = super(BalancesDueAPIView, self).paginate_queryset(
             queryset)
         return self.decorate_queryset(page if page else queryset)
+
+
+class BalancesDownloadView(CSVWriterMixin, MetricsMixin, APIView):
+
+    def retrieve_metrics(self, date_periods):
```

Comment:
It seems the `retrieve_metrics` shouldn't be redefined (as a copy/paste of the API version) but rather a `BalancesMetricsMixin` should be extracted out of `BalancesAPIView` that can be used for both, the API and Download.